### PR TITLE
Implement Contact Request flow in Contact Profile

### LINKED ISSUE #18965: Feature: New contact request user flow
Blocked by:
- #18962
- #18964

User story:

10. **Initiating a Contact Request**
    - As a user, I want to initiate a contact request to a stranger, so that I can express my interest to connect with them for future interactions within the app.

11. **Adding a Personalized Message to Contact Request**
    - As a user, I want to add a personalized message with my contact request, so that the recipient has context on who I am and why I want to connect.

12. **Reviewing Contact Request Before Sending** 
    - As a user, I want to review my contact request and personal message before sending, so that I can make sure it contains the right information and is free of errors.

13. **Sending the Contact Request**
    - As a user, I want to send the contact request after typing my message, so that I can formally request to connect with the other user.

Scope:
- UI for the full user flow
- No toast
- Create a separate issue if the auto-expand input is not already implemented

Figma: https://www.figma.com/file/jGs1P4RU4R1NRrY0xXmHC1/Profile-interactions?type=design&node-id=0-38997&mode=design&t=L27O8eaCfS59gCGl-4

Image

### PR DESCRIPTION
**Pull request by seanstrom**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #18965 
aligns with #18998 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

This PR attempts to implement the new designs for the Contact Request in Contact Profiles. This PR is dependent on the changes inside PR #18998, which means that contact requests will not support personalised messages until those changes are merged. This PR aligns with the changes inside #18998, but does directly rely on those commits in this branch.

### Testing notes

* ~~Note that while testing this PR the personalised messages will not be functioning properly until #18998 is merged.~~ All other functionality related to #18965 is intended to be functioning correctly.
  * PR #18998 is merged so personalised messages are now testable 🙌 
* Another note, this PR modifies the Quo `top-drawer` component, which may affect other UI elements like bottom sheets or drawers.

#### Platforms

- Android
- iOS

##### Functional

- New Contact Profile UI
- Top-Drawer components
- Top-Drawer components with Context-Tags

### Steps to test

- Open Status mobile and Status desktop app
- Enable the `new-contact-ui` feature flag in the mobile app
- View a contact profile on mobile
- Press the contact request button
- Provide a contact request message and press the "send contact request" button

#### Before

Screenshot 2024-02-29 at 09 28 58

#### After

https://github.com/status-im/status-mobile/assets/2845768/f88953fa-78bd-458c-8602-5666b4a59d22

status: ready

### CODE REVIEW

File: src/status_im/contexts/profile/contact/header/view.cljs [hunk 1]
```diff
@@ -31,4 +34,14 @@
...
+     [rn/view {:style style/button-wrapper}
+      [quo/button
+       {:on-press  #(rf/dispatch [:show-bottom-sheet
+                                  {:content (fn [] [contact-request/view
+                                                    {:customization-color customization-color
+                                                     :profile-picture     profile-picture
+                                                     :full-name           full-name
+                                                     :public-key          public-key}])}])
+        :icon-left :i/add-user}
+       (i18n/label :t/send-contact-request)]]]))
```
Comment (seanstrom) [inline src/status_im/contexts/profile/contact/header/view.cljs]: I'm wondering if I should subscribe directly to these values from inside the `contact-request/view` component, or just pass them along like I'm doing here. Thoughts?

File: src/quo/components/drawers/drawer_top/view.cljs [hunk 2]
```diff
@@ -76,7 +76,8 @@
    (str description " · " (i18n/label :t/on-device))])
 
 (defn- context-tag-subtitle
-  [{:keys [context-tag-type community-logo community-name account-name emoji customization-color]}]
+  [{:keys [context-tag-type community-logo community-name account-name emoji customization-color
+           full-name profile-picture]}]
```
Comment (seanstrom) [inline src/quo/components/drawers/drawer_top/view.cljs:80]: I'm not sure what's the best way to pass all these different props into this `drawer-top` component. There seems to be a large amount of potentially useful props that need to split up between all the different variants of draw-top components. In this case, I needed to add more support for props specifically for the draw-top component with the embedded context-tag component.

Any thoughts on how we could make this more maintainable in the future? Some ideas around isolating prop arguments would be nice 🙏

File: src/status_im/contexts/profile/contact/contact_request/view.cljs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+(ns status-im.contexts.profile.contact.contact-request.view
```
Comment (seanstrom) [inline src/status_im/contexts/profile/contact/contact_request/view.cljs:1]: Not sure if this is the best name organising this code, lmk if you have any suggestions 🙏

**Review by seanstrom — COMMENTED** [review]
Self Review / Comments

File: src/quo/components/drawers/drawer_top/view.cljs (hunk 2 above)
Comment (J-Son89) [reply src/quo/components/drawers/drawer_top/view.cljs:80]: I think part of the issue of components with multiple sub components to care about is the design specifications. We really need to address this at the root of the problem. The design system should highlight when it's possible that a component takes a generic child into it's structure and then we can compose the components a bit easier and have a much cleaner api.

File: src/status_im/contexts/profile/contact/header/view.cljs (hunk 1 above)
Comment (J-Son89) [reply src/status_im/contexts/profile/contact/header/view.cljs]: generally avoid passing subscription values along and prefer to subscribe inside the component using the values.

**Review by mohsen-ghafouri — APPROVED** [review]
Nice work, LGTM.

**Review by yqrashawn — APPROVED** [review]

**Comment by pavloburykh** [comment]
@seanstrom thanx for the PR! Does this PR require manual testing? Is anything could be affected in app if feature flag is disabled? Are ```Top-Drawer components``` and ```Top-Drawer components with Context-Tags``` can be affected in case if feature flag is Disabled? If yes - could you please provide examples of places in app were we should check for regression of this components. Thank you!

**Comment by seanstrom** [comment]
Hey @pavloburykh 👋 

The changes to the top-drawer component will be active without the feature flag, and I think we can find examples of it being used by looking for features that use the drawer / bottom-sheet UI. I looked through the code and it seems there's some usages that can be found while requesting to join a community. 

https://github.com/status-im/status-mobile/assets/2845768/6b58b701-7ef0-4186-8577-2a93b3cbbd64

---

There could be some more examples that are worth double-checking and I'm not sure where all of them are located, but I can help find some other places by looking through the code some more. Lmk if we need some more examples 🙏

**Comment by pavloburykh** [comment]
@seanstrom thank you! PR is ready for merge.

**Pull request merged**